File: 2023/advent_of_code/test/q13/q13_test.exs
defmodule AdventOfCode.Q13Test do
  use ExUnit.Case
  doctest AdventOfCode.Q13

  test "Q13 Part 1 Example works" do
    file_stream = File.stream!("./test/q13/q13_example.txt")
    result = AdventOfCode.Q13.part1(file_stream)
    assert result == 405
  end

  test "Q13 Part 1 works" do
    file_stream = File.stream!("./test/q13/q13.txt")
    result = AdventOfCode.Q13.part1(file_stream)
    assert result == 39939
  end

  test "Q13 Part 2 Example works" do
    file_stream = File.stream!("./test/q13/q13_example.txt")
    result = AdventOfCode.Q13.part2(file_stream)
    assert result == 400
  end

  test "Q13 Part 2 works" do
    file_stream = File.stream!("./test/q13/q13.txt")
    result = AdventOfCode.Q13.part2(file_stream)
    assert result == 32069
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q13/Q13Test.kt
package q13

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q13Test {

    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q13.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(5588, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q13.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(23958, part2(filePath!!))
    }
}